utils.js

```diff
@@ -43,10 +43,66 @@ const prefs = require('sdk/simple-prefs');
 const {get, set, reset} = require('sdk/preferences/service');
 const NS_XUL = 'http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul';
 const self = require('sdk/self');
+const main = require('main');
```

Comment:
If main requires utils, and utils requires main, what happens?
For more sanity, could we perhaps have main call a function in utils to set up the study, and pass in the `unload` function?
(I _think_ everything else in utils is a function, so this would make it more consistent, too.  😉)